LunaDll/GlobalFuncs.cpp

```diff
@@ -1037,3 +1037,15 @@ void HandleEventsWhileLoading()
         lastTime = thisTime;
     }
 }
+
+std::string GetEditorPlacedItem()
```

Comment:
This code is not thread-safe, and is liable to give bad output or crash at random if you get unlucky.
There is a race condition here, because `IPCPipeServer::ReadThread()` runs in a separate thread from the main game thread that `GetEditorPlacedItem()` would be called from. You need to use a [lock](https://en.cppreference.com/w/cpp/thread/lock_guard) and [mutex](https://en.cppreference.com/w/cpp/thread/mutex) to control access to `editorPlacedItem` and make it safe. See the C++ standard library classes I linked. You can find examples of them being used already in the LunaLua code too.